joomla/joomla-cms
# Make js ajax error messages more useful when using Joomla.request

### Summary of Changes

There are some cases we get some strange message on ajax errors. Thsi happens when, for instance, when we use the new vannila js `Joomla.request` instead of the jquery ajax methods.

It seems this happens because jQuery ajax error returns different values than plain vanilla js methods do.

This PR intends to make those messages more useful in this contexts.
### Testing Instructions
1. Code review.
2. You need to make ajax errors to test this. Example:
3. Set session time to one minute so the keepalive ajax call takes one minute.
4. Now edit any content in admin and leave the edit view open so keepalive keep running
   . add the following code in https://github.com/joomla/joomla-cms/blob/3.7.x/administrator/components/com_ajax/ajax.php#L11 to make an ajax error

``` php
$app = JFactory::getApplication();
$app->setHeader('Content-Type', 'application/json; charset=utf-8');
$app->sendHeaders();
echo jason_encode('a');
$app->close();
die();
```
- Wait for keepalive and you will get something like this
  ![image](https://cloud.githubusercontent.com/assets/9630530/19418856/be683022-93c4-11e6-88a6-e76a89665bd0.png)
- Now apply patch, clear browsers cache and repeat the same steps. You should get soemthing like this
  ![image](https://cloud.githubusercontent.com/assets/9630530/19418920/c637b948-93c5-11e6-9fac-b7b5fef41027.png)
### Documentation Changes Required

None.

@Fedik please review this, as you were the one that created the new Joomla.request method, and i could be missing something here.

@dgt41 please check too.
